jems-testing/src/main/java/org/dmfs/jems/hamcrest/matchers/StackMatcher.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.jems.hamcrest.matchers;
+
+import org.dmfs.iterables.decorators.Mapped;
+import org.dmfs.iterables.elementary.Seq;
+import org.dmfs.iterators.Function;
+import org.dmfs.jems.iterable.elementary.StackIterable;
+import org.dmfs.jems.stack.Stack;
+import org.hamcrest.FeatureMatcher;
+import org.hamcrest.Matcher;
+import org.hamcrest.Matchers;
+
+import static org.hamcrest.Matchers.is;
+
+
+/**
+ * A {@link Matcher} which matches an entire {@link Stack}.
+ *
+ * @author Marten Gajda
+ */
+public final class StackMatcher<T> extends FeatureMatcher<Stack<T>, Iterable<T>>
+{
+
+    public static <E> Matcher<Stack<E>> emptyStack()
+    {
+        return new StackMatcher<>(Matchers.<E>emptyIterable());
+    }
+
+
+    @SafeVarargs
+    public static <E> Matcher<Stack<E>> stacked(E... elements)
+    {
+        return new StackMatcher<>(IterableMatcher.iteratesTo(
+                new Mapped<>(
+                        new Seq<>(elements),
+                        new Function<E, Matcher<E>>()
```

Comment:
You can use the existing `EqualMatcherFunction` for this.

Reply:
Ah, I didn't know we have something like that. I'll update it.